tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,523 @@
+			ztest_test_skip();
+		}
+
+		ztest_test_fail();
+	}
+
+	store_data_and_factory_reset();
+
+	storage_validate_ram();
+
+	test_as_finished_mark(RESET_TEST_FINISHED_NORMAL_RESET_VALIDATE_RAM_BIT_POS);
+}
+
+static void test_normal_reset_validate_nvs(void)
+{
+	if (test_stage != RESET_TEST_STAGE_NORMAL) {
+		if (test_finished_mask &
+		    BIT(RESET_TEST_FINISHED_NORMAL_RESET_VALIDATE_NVS_BIT_POS)) {
+			ztest_test_skip();
+		}
+
+		ztest_test_fail();
+	}
+
+	store_data_and_factory_reset();
+
+	storage_validate_nvs();
+
+	test_as_finished_mark(RESET_TEST_FINISHED_NORMAL_RESET_VALIDATE_NVS_BIT_POS);
+}
+
+static void test_interrupted_reset_validate_ram(void)
+{
+	int err;
+
+	switch (test_stage) {
+	case RESET_TEST_STAGE_NORMAL:
+		storage_reset_reboot = true;
+
+		test_stage = RESET_TEST_STAGE_INTERRUPT_VALIDATE_RAM;
+		err = settings_save_one(SETTINGS_RESET_TEST_FULL_NAME(SETTINGS_STAGE_KEY_NAME),
+					&test_stage, sizeof(test_stage));
+		zassert_ok(err, "Unexpected error in test settings");
+
+		store_data_and_factory_reset();
+
+		/* The recent instruction should trigger reboot so this code should never be
+		 * entered.
+		 */
+		ztest_test_fail();
+
+		break;
+
+	case RESET_TEST_STAGE_INTERRUPT_VALIDATE_RAM:
+		storage_validate_ram();
+
+		test_as_finished_mark(RESET_TEST_FINISHED_INTERRUPTED_RESET_VALIDATE_RAM_BIT_POS);
+
+		break;
+
+	case RESET_TEST_STAGE_INTERRUPT_VALIDATE_NVS:
+		if (test_finished_mask &
+		    BIT(RESET_TEST_FINISHED_INTERRUPTED_RESET_VALIDATE_RAM_BIT_POS)) {
+			ztest_test_skip();
+		}
+
+		ztest_test_fail();
+
+		break;
+
+	default:
+		ztest_test_fail();
+		break;
+	}
+}
+
+static void test_interrupted_reset_validate_nvs(void)
+{
+	int err;
+
+	switch (test_stage) {
+	case RESET_TEST_STAGE_NORMAL:
+		/* This shouldn't happen as the test_interrupted_reset test should change
+		 * the test_stage variable.
+		 */
+		ztest_test_fail();
+
+		break;
+
+	case RESET_TEST_STAGE_INTERRUPT_VALIDATE_RAM:
+		storage_reset_reboot = true;
+
+		test_stage = RESET_TEST_STAGE_INTERRUPT_VALIDATE_NVS;
+		err = settings_save_one(SETTINGS_RESET_TEST_FULL_NAME(SETTINGS_STAGE_KEY_NAME),
+					&test_stage, sizeof(test_stage));
+		zassert_ok(err, "Unexpected error in test settings");
+
+		store_data_and_factory_reset();
+
+		/* The recent instruction should trigger reboot so this code should never be
+		 * entered.
+		 */
+		ztest_test_fail();
+
+		break;
+
+	case RESET_TEST_STAGE_INTERRUPT_VALIDATE_NVS:
+		storage_validate_nvs();
+
+		test_as_finished_mark(RESET_TEST_FINISHED_INTERRUPTED_RESET_VALIDATE_NVS_BIT_POS);
+
+		break;
+
+	default:
+		ztest_test_fail();
+		break;
+	}
+}
+
+void test_main(void)
+{
+	/* The test_interrupted_reset_validate_ram and the test_interrupted_reset_validate_nvs tests
+	 * must be placed in the suite in this exact places, because of dependencies during handling
+	 * the test_stage variable.
+	 */
+	ztest_test_suite(fast_pair_storage_factory_reset_tests,
```

Comment:
I wonder about one thing (out of curiosity), is there a reason to use [deprecated ZTest API](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/develop/test/ztest_deprecated.html)  instead of [new one](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/develop/test/ztest.html#test-framework)?

Reply:
No specific reason. I was not aware that this API is deprecated. I will use new API in that case if there will be no problems with that.

Reply:
Or at second thought we could stay with older API to be consistent with other Fast Pair tests. Later on we could switch to newer API if needed.

Reply:
Would that be ok?

Reply:
Sure, like I wrote, it was mainly out of curiosity ;)